temporal-sdk/src/main/java/io/temporal/client/WorkflowStub.java

```diff
@@ -145,22 +145,50 @@ <R> R update(
    * @param firstExecutionRunId specifies the RunID expected to identify the first run in the
    *     workflow execution chain. If this expectation does not match then the server will reject
    *     the update request with an error.
+   * @param waitPolicy specifies at what point in the update request life cycles this request should
+   *     return.
    * @param resultClass class of the update return value.
    * @param <R> type of the update return value.
-   * @param resultType type of the workflow return value. Differs from resultClass for generic
-   *     types.
+   * @param resultType type of the update return value. Differs from resultClass for generic types.
    * @param args update method arguments
-   * @return update reference that can be used to get the result of the update.
+   * @return update handle that can be used to get the result of the update.
    */
   @Experimental
   <R> UpdateHandle<R> startUpdate(
```

Comment:
I think the parameter count on this overload is large enough that you need `WorkflowUpdateOptions` + builder (like `WorkflowOptions`, `ChildWorkflowOptions`, etc). This will allow us to add options as needed too.